### PR TITLE
pidgin-osd: init at 0.1.0

### PR DESCRIPTION
**Pull request by edanaher**
###### Motivation for this change

This is a straightforward pidgin plugin; not sure how popular it is, but it's cheap to add.  I hand-rolled this plugin on Gentoo; might as well do it properly on Nix.
###### Things done
- [x] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS) - **I believe I did this correctly**; I set this (well, nix.useSandbox) in /etc/nixos/configuration.nix, ran nixos-rebuild switch, and then compiled it.
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"` - **No dependencies** on this new package
- [x] Tested execution of all binary files (usually in `./result/bin/`) - **more or less** - I built a pidgin-with-plugins using it, and it works for me.
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).  **To the best of my ability** without spending too long agonizing.

---

A straightforward pidgin plugin; kind of ancient, but still works fine.

### CODE REVIEW

**Comment by DamienCassou** [comment]
I can't make it work. Could you please explain? I've tried with:

```nix
{
pidgin = callPackage <nixpkgs/pkgs/applications/networking/instant-messengers/pidgin> {
  openssl = openssl;
  gnutls = null;
  startupnotification = libstartup_notification;
  plugins = [ pidginosd ];
};
}
```

**Comment by edanaher** [comment]
I think you have to use pidgin-with-plugins, as described by [the directions for OTR](https://nixos.org/wiki/Pidgin_with_OTR) (at least, that's what I did).  I installed to my user using nix-env, with the following in my .nixpkgs/config.nix:

```
  packageOverrides = pkgs: with pkgs; rec {
    pidgin-with-plugins = pkgs.pidgin-with-plugins.override {
      plugins = [ pidginotr pidginosd ];
    };
  };
```

File: pkgs/applications/networking/instant-messengers/pidgin-plugins/pidgin-osd/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+, autoconf, automake, libtool } :
+
+stdenv.mkDerivation rec {
+  name = "pidgin-osd-0.1.0";
+  src = fetchurl {
+    url = https://github.com/mbroemme/pidgin-osd/archive/pidgin-osd-0.1.0.tar.gz;
+    sha256 = "11hqfifhxa9gijbnp9kq85k37hvr36spdd79cj9bkkvw4kyrdp3j";
+  };
+
+  preConfigure = "./autogen.sh";
```
Comment (fpletz) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/pidgin-osd/default.nix]: It's better to use `autoreconfHook` here and remove `preConfigure` and the deps on `autoconf`, `automake`, and `libtool`. 

Just add `nativeBuildInputs = [ autoreconfHook ];`.

Comment (edanaher) [inline pkgs/applications/networking/instant-messengers/pidgin-plugins/pidgin-osd/default.nix]: Ooh, I figured there had to be something like that, but just couldn't find it.  Unfortunately, naively swapping that in gave me
```
Makefile.am: error: required file './ChangeLog' not found
```
I don't see any obvious references to ChangeLog in the project or in a couple minutes of looking at autoreconfHook.  I should have some time to dig in in the next few days.

(And for future reference, I'm guessing this should be a single commit to keep the nixpkgs history clean, and force-pushing tends to make the github conversation ugly, so I should open a new PR on a new branch with the new single commit?  Or is it okay to have two commits in the PR?)

**Comment by edanaher** [comment]
I found the problem with ChangeLog; autoreconf is running `automake --add-missing --copy --force-missing`, which appears to absolutely require a few of the standard files, like AUTHORS and ChangeLog.  And since this package doesn't have them, it fails.

My quick fix is a preAutoreconf hook to touch ChangeLog; it's kind of ugly, but works (and is still cleaner than manually reconfiguring).

**Pull request merged**